lib/cocoapods-core/specification/consumer.rb

```diff
@@ -138,9 +138,12 @@ def user_target_xcconfig
 
       # @!group Test Support
 
-      # @return [Symbol] the test type supported by this Pod.
+      # @return [Symbol] the test type supported by this specification. This will always
+      #         be `nil` for root specifications or subspecs that are not test specifications.
       #
-      spec_attr_accessor :test_type
+      def test_type
+        value_for_attribute(:test_type) unless spec.root? || !spec.test_specification?
```

Comment:
`test_specification?` should be computed based on `test_type` I think - Might be better to have the default be `:none` or `nil`?